geth/mailservice/mailservice_api.go

```diff
@@ -27,12 +36,21 @@ var (
 
 // PublicAPI defines a MailServer public API.
 type PublicAPI struct {
-	provider ServiceProvider
+	service  *MailService
+	provider ServiceProvider // just a shortcut
```

Comment:
hmm, does it make sense to make it a method instead? `provider()`.
that keeps us from ever creating a structure with an inconsistent state...